test/e2e/tests/vision/utils.ts

```diff
@@ -0,0 +1,37 @@
+import {type Page} from '@playwright/test'
+
+export function encodeQueryString(
+  query: string,
+  params: Record<string, unknown> = {},
+  options: Record<string, string | string[]> = {},
+): string {
+  const searchParams = new URLSearchParams()
+  searchParams.set('query', query)
+
+  for (const [key, value] of Object.entries(params)) {
+    searchParams.set(`$${key}`, JSON.stringify(value))
+  }
+
+  for (const [key, value] of Object.entries(options)) {
+    if (value) searchParams.set(key, `${value}`)
+  }
+
+  return `?${searchParams}`
+}
+
+export const openVisionTool = async (page: Page) => {
+  await page.goto('/test/vision')
+  // Wait for vision to be visible
+  await page.waitForSelector('[data-testid="vision-root"]', {
+    timeout: 30_000,
+  })
+}
```

Comment:
as far as I know in e2e tests for playwright it's preferrable to actually do an expect vs an await, you can see how we do it in other tests but the general is

playwright tends to be decent at expecting things to be visible and not taking too long to check. this is often enough and doesn't require a timeout :)

```suggestion
await expect(page.getByTestId('vision-root')).toBeVisible()
```

Reply:
Ohh right we have the `getByTestId` thank you!